components/datetime/src/options/mod.rs

```diff
@@ -56,6 +56,7 @@ pub mod preferences;
 /// At the moment only the [`length::Bag`] works, and we plan to extend that to support
 /// `ECMA402` like components bag later.
 #[derive(Debug, Clone)]
+#[allow(clippy::exhaustive_enums)] // this type is stable
```

Comment:
Issue: No. This is a textbook example of a struct that should definitely be non_exhaustive. We have plans to add more variants to this options bag in the future.

Reply:
Yeah I was unsure about this, but I left a comment on the issue for this PR with a broad categorization